### ISSUE TITLE
Add Pact::SomethingLike support to JsonBodyMatcher

### ISSUE THREAD

**Issue by lukestephenson** [DiUS/pact-jvm#237]
Pact supports a `SomethingLike` / type matcher. https://github.com/realestate-com-au/pact/wiki/Regular-expressions-and-type-matching-with-Pact#type-matching

This is currently not implemented in the jvm implementation.  If it was implemented, I would have expected to find it here: https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-matchers/src/main/scala/au/com/dius/pact/matchers/JsonBodyMatcher.scala#L45

I couldn't find the code for this on the rea master of pack-broker, but found something similar on a fork:
https://github.com/allavena/pact/blob/c4365c53bba6381309924a1aa3884e8d73a81f7d/lib/pact/matchers/matchers.rb#L38

**Comment by uglyog** [comment]
Thanks for the pull request.

I'm a bit hesitant to merge this, as the SomethingLike matcher is a direct serialisation of the Ruby matcher class into JSON. The main problem is this means that all implementations need to support the way Ruby serialises objects into JSON.

Instead, we created a language independent way of doing like matching that both the Ruby and JVM version does support. See https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-consumer-junit%2FREADME.md#ensuring-all-items-in-a-list-match-an-example-220 and https://github.com/DiUS/pact-jvm/tree/master/pact-jvm-consumer-groovy#ensuring-all-items-in-a-list-match-an-example-220 as well as https://github.com/realestate-com-au/pact/wiki/v2-flexible-matching in the Ruby version.

**Comment by lukestephenson** [comment]
Hi @uglyog, Thanks for sharing.  I'm finding the ruby version hard to follow, so looks like the PR I need to submit is one to improve the documentation.

This doc https://github.com/realestate-com-au/pact/wiki/v2-flexible-matching doesn't detail other matchers except for `each_like`.

**Comment by lukestephenson** [comment]
Actually, it is documented, but not where I was looking for it. https://github.com/realestate-com-au/pact/wiki/Regular-expressions-and-type-matching-with-Pact#type-matching
